test-utils/src/rpc/rpc_connection.rs

```diff
@@ -0,0 +1,59 @@
+use crate::rpc::errors::RpcError;
+use crate::transaction_params::TransactionParams;
+use account_compression::initialize_address_merkle_tree::{AnchorDeserialize, Pubkey, Rent};
+use anchor_lang::solana_program::clock::Slot;
+use anchor_lang::solana_program::instruction::Instruction;
+use solana_sdk::account::{Account, AccountSharedData};
+use solana_sdk::hash::Hash;
+use solana_sdk::signature::{Keypair, Signature};
+use solana_sdk::transaction::Transaction;
+
+pub trait RpcConnection {
+    fn create_and_send_transaction_with_event<T>(
+        &mut self,
+        instruction: &[Instruction],
+        payer: &Pubkey,
+        signers: &[&Keypair],
+        transaction_params: Option<TransactionParams>,
+    ) -> impl std::future::Future<Output = Result<Option<T>, RpcError>> + Send
+    where
+        T: AnchorDeserialize + Send;
+
+    fn create_and_send_transaction(
+        &mut self,
+        instruction: &[Instruction],
+        payer: &Pubkey,
+        signers: &[&Keypair],
+    ) -> impl std::future::Future<Output = Result<Signature, RpcError>> + Send;
+
+    fn get_payer(&self) -> &Keypair;
+    fn get_account(
+        &mut self,
+        address: Pubkey,
+    ) -> impl std::future::Future<Output = Result<Option<Account>, RpcError>> + Send;
+    fn set_account(&mut self, address: &Pubkey, account: &AccountSharedData);
+    fn get_rent(&mut self) -> impl std::future::Future<Output = Result<Rent, RpcError>> + Send;
+    fn get_latest_blockhash(
+        &mut self,
+    ) -> impl std::future::Future<Output = Result<Hash, RpcError>> + Send;
+    fn process_transaction_with_metadata(
+        &mut self,
+        transaction: Transaction,
+    ) -> impl std::future::Future<Output = Result<(), RpcError>> + Send;
```

Comment:
in program test this actually returns meta data.
What about changing it to process_transaction?
``
    fn send_transaction(
        &mut self,
        transaction: Transaction,
    ) -> impl std::future::Future<Output = Result<Signature, RpcError>> + Send;
    ``

Reply:
the equivalent in solana client
https://docs.rs/solana-client/latest/solana_client/nonblocking/rpc_client/struct.RpcClient.html#method.send_and_confirm_transaction

Reply:
I renamed  `process_transaction_with_metadata` to `process_transaction` + we already have 
 ```
 fn create_and_send_transaction(
        &mut self,
        instruction: &[Instruction],
        payer: &Pubkey,
        signers: &[&Keypair],
    ) -> impl std::future::Future<Output = Result<Signature, RpcError>> + Send;
```